platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -256,62 +179,146 @@ private static KeyStore getKeyStore(String location, String password) {
     return keyStore;
   }
 
-  private static void createSolrCore(
-      String url, String coreName, String configFileName, CloseableHttpClient httpClient)
+  private static void createSolrCore(String coreName, CloseableHttpClient httpClient)
       throws IOException, SolrServerException {
 
+    String solrUrl = getSolrUrl();
     try (CloseableHttpClient closeableHttpClient = httpClient; // to make sure it gets closed
         HttpSolrClient client =
             (httpClient != null
-                ? new HttpSolrClient.Builder(url).withHttpClient(httpClient).build()
-                : new HttpSolrClient.Builder(url).build())) {
+                ? new HttpSolrClient.Builder(solrUrl).withHttpClient(httpClient).build()
+                : new HttpSolrClient.Builder(solrUrl).build())) {
 
-      HttpResponse ping = client.getHttpClient().execute(new HttpHead(url));
+      HttpResponse ping = client.getHttpClient().execute(new HttpHead(solrUrl));
       if (ping != null && ping.getStatusLine().getStatusCode() == 200) {
         ConfigurationFileProxy configProxy =
             new ConfigurationFileProxy(ConfigurationStore.getInstance());
         configProxy.writeSolrConfiguration(coreName);
         if (!solrCoreExists(client, coreName)) {
           LOGGER.debug("Solr({}): Creating Solr core", coreName);
-
-          String configFile = StringUtils.defaultIfBlank(configFileName, DEFAULT_SOLRCONFIG_XML);
-          String solrDir;
-
-          if (AccessController.doPrivileged(
-              (PrivilegedAction<Boolean>) () -> System.getProperty(SOLR_DATA_DIR) != null)) {
-            solrDir =
-                AccessController.doPrivileged(
-                    (PrivilegedAction<String>) () -> System.getProperty(SOLR_DATA_DIR));
-          } else {
-            solrDir =
-                Paths.get(
-                        AccessController.doPrivileged(
-                            (PrivilegedAction<String>) () -> System.getProperty("karaf.home")),
-                        "data",
-                        "solr")
-                    .toString();
-          }
-
-          String instanceDir = Paths.get(solrDir, coreName).toString();
-
-          String dataDir = Paths.get(instanceDir, "data").toString();
-
+          String dataDir = getCoreDataDir(coreName);
```

Comment:
❗️ This also internally calls `getCoreDir()` which means we are going through the access controller multiple times for the same info.